components/ScoreCard.js

```diff
@@ -14,7 +16,15 @@ const ScoreCard = ({ id, playedAt, totalScore, userId, userName }) => {
           {playedAt}
         </div>
         <div>
-          {`${userName} posted a score of ${totalScore}`}
+          {useRouter().pathname !== '/golfers/[id]' && (
+            <Link href={`/golfers/${encodeURIComponent(userId)}`}>
+              <a className="hover:underline text-blue-700">{userName}</a>
+            </Link>
+          )}
+          {useRouter().pathname === '/golfers/[id]' && (
+            <span>{`${userName}`}</span>
+          )}
```

Comment:
Avoid calling a hook multiple times if you don't need too.

Move the result of `useRouter()` to a constant/variable and use that one to get the path name.